src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +956,738 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+                    IValueNode node = GraphQLUtils.GetFieldNodeForGivenFieldName(parameterNodes, referencedEntity.Item1);
+                    PerformDbInsertOperation(context, node.Value, sqlMetadataProvider, ReferencedEntityMultipleCreateStructure, primaryKeysOfCreatedItem);
+                }
+
+                SourceDefinition currentEntitySourceDefinition = sqlMetadataProvider.GetSourceDefinition(entityName);
+                DatabaseObject entityObject = sqlMetadataProvider.EntityToDatabaseObject[entityName];
+                string entityFullName = entityObject.FullName;
+
+                // Populate the relationship fields values for the current entity.
+                if (currentEntitySourceDefinition.SourceEntityRelationshipMap.TryGetValue(entityName, out RelationshipMetadata? relationshipData) && relationshipData is not null)
+                {
+                    foreach ((string relatedEntityName, List<ForeignKeyDefinition> fkDefinitions) in relationshipData.TargetEntityToFkDefinitionMap)
+                    {
+                        DatabaseObject relatedEntityObject = sqlMetadataProvider.EntityToDatabaseObject[relatedEntityName];
+                        string relatedEntityFullName = relatedEntityObject.FullName;
+                        ForeignKeyDefinition fkDefinition = fkDefinitions[0];
+                        if (string.Equals(fkDefinition.Pair.ReferencingDbTable.FullName, entityFullName, StringComparison.OrdinalIgnoreCase)
+                            && string.Equals(fkDefinition.Pair.ReferencedDbTable.FullName, relatedEntityFullName, StringComparison.OrdinalIgnoreCase))
+                        {
+                            int count = fkDefinition.ReferencingColumns.Count;
+                            for (int i = 0; i < count; i++)
+                            {
+                                string referencingColumnName = fkDefinition.ReferencingColumns[i];
+                                string referencedColumnName = fkDefinition.ReferencedColumns[i];
+
+                                if (multipleCreateStructure.CurrentEntityParams!.ContainsKey(referencingColumnName))
+                                {
+                                    continue;
+                                }
+
+                                if (primaryKeysOfCreatedItem.TryGetValue(relatedEntityName, out Dictionary<string, object?>? relatedEntityPKs)
+                                     && relatedEntityPKs is not null
+                                     && relatedEntityPKs.TryGetValue(referencedColumnName, out object? relatedEntityPKValue)
+                                     && relatedEntityPKValue is not null)
+                                {
+                                    multipleCreateStructure.CurrentEntityParams.Add(referencingColumnName, relatedEntityPKValue);
+                                }
+                                else if (multipleCreateStructure.HigherLevelEntityPKs is not null
```

Comment:
This looks like an entirely different concern. We are looping through the relationships for which current entity is the parent and the referencing entity. In this IF condition, though the current entity is the referencing entity, it is not the parent entity but is the child entity. There can be an additional loop which adds to the `CurrentEntityParams`, all the values from the `HigherLevelEntityPKs`.